js/ui/popover.js

```diff
@@ -69,7 +70,12 @@ const getEventDelay = function(that, optionName) {
 const attachEvent = function(that, name) {
     const target = that.option('target');
     const isSelector = isString(target);
-    const event = getEventName(that, name + 'Event');
+    let event = getEventName(that, name + 'Event');
+
+    if(that.option('shading') && name === 'hide' && that.option('hideEvent') && event !== 'click' && event !== 'dxclick') {
```

Comment:
event = this.option('hideEvent')?.name, right? Seems like condition `that.option('hideEvent')` is excess

Reply:
why are `event !== 'click' && event !== 'dxclick'` conditions not tests?

Reply:
Done